src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,82 @@
+            String csv = bufferedReader.readLine();
+            while (csv != null) {
+                count++;
+                csv = bufferedReader.readLine();
+            }
+            } catch (IOException e) {
+            throw new RuntimeException("Can't process file", e);
+        }
+
+            try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String[] products = new String[count];
+            int i = 0;
+            String csv1 = reader.readLine();
+            while (csv1 != null) {
+                    products[i] = csv1;
+                    i ++;
+                    csv1 = reader.readLine();
+                }
+            return products;
+
+            } catch (IOException e) {
+            throw new RuntimeException("Can't process file", e);
+        }
+    }
+
+    private int[] calculate(String[] products) {
+        int supply = 0;
+        int buy = 0;
+        int result = 0;
+        for(String product : products) {
+            String[] sep = product.split(",");
+            if (sep[INDEX].equals(SUPPLY)) {
+                supply += Integer.parseInt(sep[VALUE]);
+            } else {
+                buy += Integer.parseInt(sep[VALUE]);
+            }
+        }
+        result = supply - buy;
+        int[] totals = {supply, buy, result};
+        return totals;
+    }
 
+    private String getStringResult(int[] totals) {
+        String calculated = SUPPLY + "," + totals[0] + System.lineSeparator()
+                + BUY + "," + totals[1] + System.lineSeparator()
+                + RESULT + "," + totals[2] + System.lineSeparator();
+        return calculated;
+    }
+
+    private void writeResult(String calculated, String toFileName) {
+    try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+        writer.write(calculated);
+    } catch (IOException e) {
+        throw new RuntimeException(e);
```

This violates checklist item #11: 'Don't ignore exceptions'. Re-throwing the exception is good, but you should also include a descriptive message to explain what went wrong, just as you did in the `getFile` method.

For example: `throw new RuntimeException("Can't write data to file: " + toFileName, e);`